src/Network/Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorTestGroupObjectCommand.cs

```diff
@@ -0,0 +1,205 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AutoMapper;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkWatcherConnectionMonitorTestGroupObject", SupportsShouldProcess = true), OutputType(typeof(PSNetworkWatcherConnectionMonitorTestGroupObject))]
+    public class NetworkWatcherConnectionMonitorTestGroupObjectCommand : ConnectionMonitorBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The test group name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The list of test configuration.")]
+        [ValidateNotNullOrEmpty]
+        public List<PSNetworkWatcherConnectionMonitorTestConfigurationObject> TestConfiguration { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The list of source endpoints.")]
+        [ValidateNotNullOrEmpty]
+        public List<PSNetworkWatcherConnectionMonitorEndpointObject> Source { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The list of destination endpoints.")]
+        [ValidateNotNullOrEmpty]
+        public List<PSNetworkWatcherConnectionMonitorEndpointObject> Destination { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The disable flag.")]
+        public SwitchParameter Disable { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            Validate();
+
+            PSNetworkWatcherConnectionMonitorTestGroupObject testGroup = new PSNetworkWatcherConnectionMonitorTestGroupObject()
+            {
+                Name = this.Name,
+                Disable = this.Disable? true:false,
+                TestConfigurations = this.TestConfiguration,
+                Sources = this.Source,
+                Destinations = this.Destination
+            };
+
+            WriteObject(testGroup);
+        }
+
+        public bool Validate()
+        {            
+            if (!this.TestConfiguration.Any())
```

Comment:
Then this parameter should be required, with ValidateNotNullOrEmpty =- in which case this is unnecessary